x-pack/plugins/fleet/public/applications/fleet/sections/agents/components/agent_enrollment_flyout/index.tsx

```diff
@@ -16,72 +16,26 @@ import {
   EuiFlexGroup,
   EuiFlexItem,
   EuiButtonEmpty,
-  EuiButton,
   EuiFlyoutFooter,
   EuiTab,
   EuiTabs,
-  EuiCallOut,
-  EuiLink,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { i18n } from '@kbn/i18n';
 
 import { useGetSettings, useUrlModal } from '../../../../hooks';
-import type { AgentPolicy } from '../../../../types';
 
 import { ManagedInstructions } from './managed_instructions';
 import { StandaloneInstructions } from './standalone_instructions';
+import { MissingFleetServerHostCallout } from './missing_fleet_server_host_callout';
+import type { BaseProps } from './types';
 
-interface Props {
+interface Props extends BaseProps {
   onClose: () => void;
-  agentPolicies?: AgentPolicy[];
 }
 
-const MissingFleetServerHostCallout: React.FunctionComponent = () => {
-  const { setModal } = useUrlModal();
-  return (
-    <EuiCallOut
-      title={i18n.translate('xpack.fleet.agentEnrollment.missingFleetHostCalloutTitle', {
-        defaultMessage: 'Missing URL for Fleet Server host',
-      })}
-    >
-      <FormattedMessage
-        id="xpack.fleet.agentEnrollment.missingFleetHostCalloutText"
-        defaultMessage="A URL for your Fleet Server host is required to enroll agents with Fleet. You can add this information in Fleet Settings. For more information, see the {link}."
-        values={{
-          link: (
-            <EuiLink
-              href="https://www.elastic.co/guide/en/fleet/current/index.html"
-              target="_blank"
-              external
-            >
-              <FormattedMessage
-                id="xpack.fleet.agentEnrollment.missingFleetHostGuideLink"
-                defaultMessage="Fleet User Guide"
-              />
-            </EuiLink>
-          ),
-        }}
-      />
-      <EuiSpacer size="m" />
-      <EuiButton
-        fill
-        iconType="gear"
-        onClick={() => {
-          setModal('settings');
-        }}
-      >
-        <FormattedMessage
-          id="xpack.fleet.agentEnrollment.fleetSettingsLink"
-          defaultMessage="Fleet Settings"
-        />
-      </EuiButton>
-    </EuiCallOut>
-  );
-};
-
 export const AgentEnrollmentFlyout: React.FunctionComponent<Props> = ({
   onClose,
+  agentPolicy,
```

Comment:
It seems odd that we have different sets of behavior for a singular agent policy and multiple agent policies. Is this is a necessity based on the designs, or is this something we could eliminate with logic such that "if one or more agent policies is provided, the form displays in X way"?

Reply:
Good point! My concern with👆🏻 approach is the unintended side-effect of thinking we are passing in 1+ policies to `agentPolicies` when we are not. 

The current API gives us the ability to say "this policy is the selected policy for the context": the flyout presentation maps directly from the props.

If we don't want to have this ability for the flyout I am happy to change and defer to your call with respect to what is best for the UX here: i.e., is it OK to have the flyout dynamically present based on count in all situations?

Reply:
I think your reasoning for this approach is sound, and I don't have the answer for whether we can make that assumption about the UX here. I think this pattern allows to effectively account for all cases, so all is good with me 👍